ui/text_selection.js

```diff
@@ -164,12 +164,12 @@ shaka.ui.TextSelection = class extends shaka.ui.SettingsMenu {
    * @private
    */
   async onTextTrackSelected_(track) {
-    // setTextTrackVisibility should be called after selectTextLanguage.
-    // selectTextLanguage sets a text stream, and setTextTrackVisiblity(true)
+    // setTextTrackVisibility should be called after selectTextTrack.
+    // selectTextTrack sets a text stream, and setTextTrackVisiblity(true)
     // will set a text stream if it isn't already set. Consequently, reversing
     // the order of these calls makes two languages display simultaneously
     // if captions are turned off -> on in a different language.
-    this.player.selectTextLanguage(track.language, track.roles[0]);
+    this.player.selectTextTrack(track);
```

Comment:
I personally think this should be selectTextTrack so it actually selects the text track that the user intends to select rather than `selectTextLanguage`, which finds the first track that has the language provided. This is because often times, CEA-608 and CEA-708 tracks may have the same language, so there may be ambiguity. 

TLDR for the sake of reliability and removing ambiguity in our UI library, I've done this.